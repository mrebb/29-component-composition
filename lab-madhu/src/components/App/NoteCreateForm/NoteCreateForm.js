import React, { Component, Fragment } from 'react';
import uuid from 'uuid/v4';

export default class NoteForm extends Component {
  constructor(props){
    super(props);
    this.state = {
      editing: false,
      completed: false,
      content: '',
      title: '',
    };
  }
  onSubmit = event => {
    event.preventDefault();
    this.props.onSubmit({ ...this.state, id: uuid() });
    this.setState({ title: '', content: '' });
  };
  onChange = event => {
    let changedVal =
      event.target.type === 'checkbox'
        ? event.target.checked
        : event.target.value;
    const changedBit = { [event.target.name]: changedVal };
    this.setState(changedBit);
  };
  render() {
    return (
      <Fragment>
        <form onSubmit={this.onSubmit} onChange={this.onChange}>
          <input
            name="title"
            value={this.state.title}
            type="text"
            placeholder="title"
          />
          <textarea
            name="content"
            value={this.state.content}
            placeholder="content"
          />
          <label>
            <span>editing</span>
            <input name="editing" type="checkbox" />
          </label>
          <label>
            <span>completed</span>
          </label>
          <input name="completed" type="checkbox" />
          <button>create note</button>
        </form>
      </Fragment>
    );
  }
}
