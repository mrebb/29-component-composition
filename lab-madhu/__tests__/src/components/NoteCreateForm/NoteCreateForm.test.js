import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import NoteCreateForm from '../../../../src/components/app/NoteCreateForm/NoteCreateForm';

describe('NotesCreateForm component', () => {
  it('is alive at application start', () => {
    let notesForm = shallow(<NoteCreateForm/>);
    expect(notesForm.find('form.notesForm').exists()).toBe(true);
  });
  it('renders a title input', () => {
    expect(shallow(<NoteCreateForm/>).find('#title').length).toEqual(1);
  });
  it('renders a content input', () => {
    expect(shallow(<NoteCreateForm/>).find('#content').length).toEqual(1);
  });
  describe('Title input', () => {
  
    it('should respond to change event and change the state of the NoteCreateForm Component', () => {
     
      const notesForm = shallow(<NoteCreateForm />);
      notesForm.find('#title').simulate('change', {target: {name: 'title', value: 'fruits'}});
     
      expect(notesForm.state('title')).toEqual('fruits');
    });
  });
   
  describe('Content input', () => {
    
    it('should respond to change event and change the state of the NoteCreateForm Component', () => {
     
      const notesForm = shallow(<NoteCreateForm />);
      notesForm.find('#content').simulate('change', {target: {name: 'content', value: 'buy fruits from store'}});
     
      expect(notesForm.state('content')).toEqual('buy fruits from store');
    });
  });
});


describe('<NotesCreateForm/> (Snapshot Test)', () => {
  it('renders right', () => {
    const component = renderer.create(
      <NoteCreateForm />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

  });
});
