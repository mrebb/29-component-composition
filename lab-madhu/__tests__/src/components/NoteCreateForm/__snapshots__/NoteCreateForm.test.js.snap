// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotesCreateForm/> (Snapshot Test) renders right 1`] = `
Array [
  <div>
    <a
      href="/"
    >
      Back to home
    </a>
  </div>,
  <div
    className="notesDiv"
  >
    <form
      className="notesForm"
      onSubmit={[Function]}
    >
      <h1>
        Create new notes
      </h1>
      <label
        htmlFor="title"
      >
        Title:  
      </label>
      <input
        id="title"
        name="title"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <br />
      <br />
      <label
        htmlFor="content"
      >
        Content:  
      </label>
      <textarea
        id="content"
        name="content"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <br />
      <br />
      <button>
        Create
      </button>
    </form>
  </div>,
]
`;
